tesseract_task_composer/core/src/task_composer_plugin_factory.cpp

```diff
@@ -282,17 +282,42 @@ std::string TaskComposerPluginFactory::getDefaultTaskComposerNodePlugin() const
 std::unique_ptr<TaskComposerExecutor>
 TaskComposerPluginFactory::createTaskComposerExecutor(const std::string& name) const
 {
-  const auto& executor_plugin_info = impl_->executor_plugin_info;
-  auto cm_it = executor_plugin_info.plugins.find(name);
-  if (cm_it == executor_plugin_info.plugins.end())
+  try
```

Comment:
My only concern with this change is that the plugins are not intended to be reconfigurable after being created which can lead to incorrect usage. Instead what if this method generated a meta package (which needs to be defined) that this function produces. Once option I explored in the past was leveraging a yaml file which include this information which could also be used for the yaml editor in tesseract_qt.

Reply:
Here is the schema I was exploring at the time. I think it would be good to discuss a little more because this could benefit all plugins throughout tesseract. 


https://github.com/tesseract-robotics/tesseract_planning/blob/master/tesseract_task_composer/config/task_composer_plugins.yaml.schema